### PR TITLE
Add PerUserKeyring#GetEncryptionKeyBySeqno

### PR DESCRIPTION
**Pull request by mlsteele**
Add `PerUserKeyring#GetEncryptionKeyBySeqno` which simplifies unboxing a teambox a bit and saves a user load.

### CODE REVIEW

File: go/engine/revoke.go [hunk 1]
```diff
@@ -193,11 +193,15 @@ func (e *RevokeEngine) Run(ctx *Context) error {
...
 	// Push the per-user-key sig
+
+	// Seqno when the per-user-key will be signed in.
+	var newPukSeqno keybase1.Seqno
 	if e.G().Env.GetSupportPerUserKey() && addingNewPUK {
 		sig1, err := libkb.PerUserKeyProofReverseSigned(me, *newPukSeed, newPukGeneration, sigKey)
 		if err != nil {
 			return err
 		}
+		newPukSeqno = me.GetSigChainLastKnownSeqno()
```
Comment (patrickxb) [inline go/engine/revoke.go]: Do you need to add 1 here?

File: go/teams/teams.go [hunk 2]
```diff
@@ -175,13 +150,10 @@ func (t *Team) perUserEncryptionKey(ctx context.Context) (*libkb.NaclDHKeyPair,
...
 		return nil, err
 	}
-	encKey, err := kr.GetEncryptionKeyByKID(ctx, encKID)
+	encKey, err := kr.GetEncryptionKeyBySeqno(ctx, t.Box.PerUserKeySeqno)
 	if err != nil {
 		return nil, err
 	}
-	if encKey.Private == nil {
-		return nil, errors.New("per user enckey is locked")
-	}
```
Comment (patrickxb) [inline go/teams/teams.go]: is there a reason for removing this check?

Comment (mlsteele) [inline go/teams/teams.go]: It should be impossible, the keyring's job is to store the secret. But checks are nice, I put it back.

File: go/engine/revoke.go (hunk 1 above)
Comment (mlsteele) [inline go/engine/revoke.go]: `PerUserKeyProofReverseSigned` modifies the `User` and bumps its sigchain. Which is also how when you do local proofs `kbsig` knows to use the new seqno and hash. This is also the reason this line must happen before `makeRevokeSig`.

Comment (mlsteele) [inline go/engine/revoke.go]: Also I just snooped the logs of a run of `TestRevokeDevicePUK` and it looks right.

File: go/teams/teams.go (hunk 2 above)
Comment (patrickxb) [inline go/teams/teams.go]: Ok, I assume I put it there because of a crash...

File: go/engine/revoke.go (hunk 1 above)
Comment (patrickxb) [inline go/engine/revoke.go]: ok, thanks.

File: go/teams/teams.go (hunk 2 above)
Comment (mlsteele) [inline go/teams/teams.go]: eek. Let me know if you see it again

**Pull request merged**